### PR TITLE
Purecap kernel kernel map allocator changes

### PR DESCRIPTION
**Pull request by qwattash**

### CODE REVIEW

**Review by brooksdavis — APPROVED** [review]

File: sys/vm/vm_kern.c [hunk 1]
```diff
@@ -291,18 +303,22 @@ kmem_alloc_attr_domainset(struct domainset *ds, vm_size_t size, int flags,
...
 	vm_object_t object;
-	vm_offset_t addr, offset, tmp;
+	vm_pointer_t addr;
+	vm_offset_t offset, tmp;
 	vm_page_t end_m, m;
 	u_long npages;
 	int pflags;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+        size = CHERI_REPRESENTABLE_LENGTH(size);
```
Comment (jrtc27) [inline sys/vm/vm_kern.c:320]: Indentation

File: sys/vm/vm_kern.c [hunk 2]
```diff
@@ -143,15 +143,17 @@ extern void     uma_startup2(void);
  *	its use, typically with pmap_qenter().  Any attempt to create
  *	a mapping on demand through vm_fault() will result in a panic. 
  */
-vm_offset_t
+vm_pointer_t
 kva_alloc(vm_size_t size)
 {
-	vm_offset_t addr;
+	vmem_addr_t addr;
```
Comment (jrtc27) [inline sys/vm/vm_kern.c:149]: This should go upstream.

File: sys/vm/vm_kern.c [hunk 3]
```diff
@@ -209,17 +211,21 @@ kmem_alloc_contig_pages(vm_object_t object, vm_pindex_t pindex, int domain,
...
 	vm_object_t object;
-	vm_offset_t addr, i, offset;
+	vm_pointer_t addr;
+	vm_offset_t i, offset;
 	vm_page_t m;
 	int pflags;
 	vm_prot_t prot;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+        size = CHERI_REPRESENTABLE_LENGTH(size);
```
Comment (jrtc27) [inline sys/vm/vm_kern.c:227]: Indentation

File: sys/vm/vm_kern.c (hunk 2 above)
Comment (qwattash) [reply sys/vm/vm_kern.c:149]: Split into separate commit

**Pull request merged**